From: Joerg Roedel <jroedel@suse.de>
Date: Tue, 26 Mar 2013 00:03:21 +0000
Subject: mlx4: Check for assigned VFs before disabling SR-IOV
Git-commit: None; This patch might change in the future
Patch-mainline: No
References: bsc#927355

This patch adds a check to the mlx4 driver to prevent
disabling SR-IOV if assigned VFs are found.

Signed-off-by: Joerg Roedel <jroedel@suse.de>
---
 drivers/net/ethernet/mellanox/mlx4/main.c |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

--- a/drivers/net/ethernet/mellanox/mlx4/main.c
+++ b/drivers/net/ethernet/mellanox/mlx4/main.c
@@ -2410,9 +2410,13 @@
 	int active_vfs = 0;
 
 	if (dev) {
-		/* Disabling SR-IOV is not allowed while there are active vf's */
+		/*
+		 * Disabling SR-IOV is not allowed while there are active or
+		 * assigned VF's
+		 */
 		if (mlx4_is_master(dev)) {
-			active_vfs = mlx4_how_many_lives_vf(dev);
+			active_vfs = max(mlx4_how_many_lives_vf(dev),
+					 pci_vfs_assigned(pdev));
 			if (active_vfs) {
 				pr_warn("Removing PF when there are active VF's !!\n");
 				pr_warn("Will not disable SR-IOV.\n");
